File: LevelSDKExampleApp/app/src/main/java/com/theshopatvsp/levelandroidsdk/ble/model/DeviceIdManager.java
package com.theshopatvsp.levelandroidsdk.ble.model;

/**
 * Created by andrco on 6/11/16.
 */
public class DeviceIdManager {
    private int packetIdOut = 0;
    private int packetIdIn = -1;

    public int getPacketIdOut() {
        return packetIdOut;
    }

    public void incPacketIdOut() {
        this.packetIdOut++;

        if( this.packetIdOut > 255 ) {
            this.packetIdOut = 0;
        }
    }

    public int getPacketIdIn() {
        return packetIdIn;
    }

    public void setPacketIdIn(int packetIdIn) {
        this.packetIdIn = packetIdIn;
    }

    public void incPacketIdIn() {
        this.packetIdIn++;

        if (this.packetIdIn > 255) {
            this.packetIdIn = 0;
        }
    }

    public void reset() {
        this.packetIdIn = -1;
        this.packetIdOut = 0;
    }
}

File: LevelIOSSdkExample/LevelIOSSdkExample/LevelIOSSdk/DeviceIdManager.swift
//
//  DeviceIdManager.swift
//  LevelIOSSDK
//
//  Created by Andrew Cook on 6/28/16.
//  Copyright © 2016 TheShop. All rights reserved.
//

import Foundation

class DeviceIdManager {
    var packetIdOut: Int = 0
    var packetIdIn: Int = -1
    
    func incPacketIdOut() {
        self.packetIdOut += 1
        
        if self.packetIdOut > 255 {
            self.packetIdOut = 0
        }
    }
    
    func incPacketIdIn() {
        self.packetIdIn += 1
        
        if self.packetIdIn > 255 {
            self.packetIdIn = 0
        }
    }
    
    func reset() {
        self.packetIdIn = -1
        self.packetIdOut = 0
    }
}